packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -91,6 +91,17 @@ function areHookInputsEqual(
   return true;
 }
 
+function shouldRunEffectsBasedOnInputs(
+  before: Array<mixed>,
+  after: Array<mixed> | void | null,
+) {
+  if (after == null || before.length !== after.length) {
+    return true;
+  }
+
+  return before.some((value, i) => (after: any)[i] !== value);
```

Comment:
flow makes me `: any` here as it thinks `after` can be `null` (despite the fact the function would have returned true already if that were the case.